slsim/Sources/galaxies.py

```diff
@@ -76,26 +119,140 @@ def draw_source(self):
+                    )
+                    galaxy["e1_1"] = e1_1
+                    galaxy["e1_2"] = e1_2
+                elif "a1" in galaxy.colnames and "b1" in galaxy.colnames:
+                    axis_ratio_1 = axis_ratio(a=galaxy["a1"], b=galaxy["b1"])
+                    ellip_1 = eccentricity(q=axis_ratio_1)
+                    e1_1, e1_2 = galaxy_projected_eccentricity(float(ellip_1))
+                    galaxy["e1_1"] = e1_1
+                    galaxy["e1_2"] = e1_2
+                else:
+                    raise ValueError(
+                        "ellipticity or semi-major and semi-minor axis are missing for"
+                        "the second light profile in galaxy_list columns"
+                    )
+            if galaxy["angular_size0"] == -1 or galaxy["angular_size1"] == -1:
+                if "a0" in galaxy.colnames and "b0" in galaxy.colnames:
+                    galaxy["angular_size0"] = average_angular_size(
+                        a=galaxy["a0"], b=galaxy["b0"]
+                    )
+                else:
+                    raise ValueError(
+                        "semi-major and semi-minor axis are missing for the first light"
+                        "profile in galaxy_list columns"
+                    )
+                if "a1" in galaxy.colnames and "b1" in galaxy.colnames:
+                    galaxy["angular_size1"] = average_angular_size(
+                        a=galaxy["a1"], b=galaxy["b1"]
+                    )
+                else:
+                    raise ValueError(
+                        "semi-major and semi-minor axis are missing for the second"
+                        "light profile in galaxy_list columns"
+                    )
+            if galaxy["n_sersic_0"] == -1 or galaxy["n_sersic_1"] == -1:
+                galaxy["n_sersic_0"] = 1
+                galaxy["n_sersic_1"] = 4
+        else:
+            raise ValueError(
+                "Provided number of light profiles is not supported. It should be"
+                "either 'single or 'double' "
+            )
         return galaxy
 
 
-def galaxy_projected_eccentricity(ellipticity):
+def galaxy_projected_eccentricity(ellipticity, rotation_angle=None):
     """Projected eccentricity of elliptical galaxies as a function of other deflector
     parameters.
 
     :param ellipticity: eccentricity amplitude
     :type ellipticity: float [0,1)
+    :param rotation_angle: rotation angle of the major axis of elliptical galaxy in
+        radian. The reference of this rotation angle is +Ra axis i.e towards the East
+        direction.
     :return: e1, e2 eccentricity components
     """
+    if rotation_angle is None:
+        phi = np.random.uniform(0, np.pi)
+    else:
+        phi = rotation_angle
     e = param_util.epsilon2e(ellipticity)
-    phi = np.random.uniform(0, np.pi)
     e1 = e * np.cos(2 * phi)
     e2 = e * np.sin(2 * phi)
     return e1, e2
+
+
+def convert_to_slsim_convention(
```

Comment:
perhaps have in the name of this function scotch as well?

Reply:
For this time, I keep a general name. In future, we might split this function for different catalog.